examples/postgresql/assembly/index.ts

```diff
@@ -0,0 +1,34 @@
+import { postgresql } from "@hypermode/functions-as";
+
+export function queryPeople(): string {
+  return postgresql.query("neon", "select * from people", []);
+}
+
+export function getPerson(name: string): string {
+  const query = "select * from people where name = $1";
+  return postgresql.query("neon", query, [name]);
+}
+
+export function addPerson(
+  name: string,
+  age: i32,
+  balance: f64,
+  privkey: i8[],
+  male: boolean,
+  lat: f64,
+  lon: f64,
+): void {
+  const query =
+    "insert into people (name, age, balance, privkey, created_at, male, home)" +
+    " values ($1, $2, $3, $4, $5, $6, $7)";
+
+  postgresql.query("neon", query, [
+    name,
+    age.toString(),
+    balance.toString(),
+    privkey.toString(),
+    new Date(Date.now()).toUTCString(),
+    male.toString(),
+    "(" + lat.toString() + "," + lon.toString() + ")",
+  ]);
```

Comment:
This isn't great.  The user shouldn't have to turn everything into a string.

It also means that we're passing all parameters as strings into Postgres, such as:

```json
["John","42,123.45","[1,2,3,4]","2024-06-19T23:59:59.999Z","true","(123.45,-67.89)"]
```

The existing `QueryVariables` class could be re-purposed here, which would turn this code into something like:

```ts
  const params = new QueryVariables();
  params.set("$1", name);
  params.set("$2", age);
  params.set("$3", balance);
  params.set("$4", privkey);
  params.set("$5", new Date(Date.now()));
  params.set("$6", male);
  params.set("$7", `(${lat},${lon})`);

  postgresql.query("neon", query, params);
```

However, since postgres only use positional parameters (unlike SQL Server and others that use named parameters), we should make a new class in our library:

```ts
export class QueryParameters {
  private data: string[] = [];

  public add<T>(value: T): void {
    this.data.push(JSON.stringify(value));
  }

  public toJSON(): string {
    return `[${this.data.join(",")}]`;
  }
}
```

Then the usage can be:

```ts
  const params = new QueryParameters();
  params.add(name);
  params.add(age);
  params.add(balance);
  params.add(privkey);
  params.add(new Date(Date.now()));
  params.add(male);
  params.add(`(${lat},${lon})`);

  postgresql.query("neon", query, params);
```

The resulting JSON array would then not contain only strings, but for example:

```json
["John",42,123.45,[1,2,3,4],"2024-06-19T23:59:59.999Z",true,"(123.45,-67.89)"]
```

Reply:
Also you could try passing the lat/lon as an array - I'm not sure how Postges needs it though.

```ts
params.add(<f32[]>[lat, lon]);
```